walk-in-interview/src/main/java/com/google/job/data/JobsDatabase.java

```diff
@@ -238,6 +249,109 @@
+
+        DocumentReference docRef = applicantAccountsCollection.document(applicantId);
+
+        return ApiFutures.transform(
+            docRef.get(),
+            documentSnapshot -> {
+                if (!documentSnapshot.exists()) {
+                    throw new IllegalArgumentException("Invalid applicantId");
+                }
+
+                List<String> interestedList = (List<String>) documentSnapshot.get(INTERESTED_JOBS_FIELD);
+
+                List<Job> jobList = fetchAllJobsFromIds(interestedList);
+                // TODO(issue/34): adjust range/total count based on pagination
+                long totalCount = jobList.size();
+                Range<Integer> range = Range.between(1, jobList.size());
+
+                return new JobPage(jobList, totalCount, range);
+            },
+            MoreExecutors.directExecutor()
+        );
+    }
+
+    /**
+     * This will return a list of jobs given the list of jobIds.
+     * It will call on a helper function that will process the list at max 10 ids at a time.
+     * Any jobs that are not active or invalid will not be included in the list returned.
+     *
+     * @param jobIds The list of jobIds.
+     * @return The list of jobs.
+     */
+    public List<Job> fetchAllJobsFromIds(List<String> jobIds) {
+        ImmutableList.Builder<Job> jobListBuilder = ImmutableList.builder();
+        try {
+            int counter = 0;
+            // TODO(issue/34): pagination could also be included here.
+            while (counter + FIRESTORE_IN_QUERY_MAX_ARGS < jobIds.size()) {
+                List<String> subList = jobIds.subList(/* inclusive */ counter, /* exclusive */ counter + 10);
+                List<Job> fetchedList = fetchJobsFromIds(subList).get(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+                jobListBuilder.addAll(fetchedList);
+                counter = counter + FIRESTORE_IN_QUERY_MAX_ARGS;
+            }
+            List<String> subList = jobIds.subList(/* inclusive */ counter, /* exclusive */ jobIds.size());
+            List<Job> fetchedList = fetchJobsFromIds(subList).get(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+            jobListBuilder.addAll(fetchedList);
+        } catch (IOException | InterruptedException | ExecutionException | TimeoutException e) {
+            log.log(Level.SEVERE, "error while getting interested job list ", e);
+        }
+
+        return jobListBuilder.build();
+    }
+
+    /**
+     * This will return a future of a list of jobs given the list of jobIds.
+     * Any jobs that are not active or invalid will not be included in the list returned.
+     * This only handles jobIds of max 10 at a time.
+     *
+     * @param jobIds The list of jobIds.
+     * @return Future of the list of jobs.
+     */
+    private Future<List<Job>> fetchJobsFromIds(List<String> jobIds) throws IOException {
+        CollectionReference jobsCollection = FireStoreUtils.getFireStore().collection(JOB_COLLECTION);
```

Comment:
FYI you should check if the list is empty and immediately return empty list back.